### PR TITLE
Clone Verif Environment - Script

### PR DESCRIPTION
**Pull request by silabs-robin**
This PR adds a script to clone the external 40x verif environment subtree repo.

It is clean, simple, and extensible to other cores.

I can rewrite this in Python or make if we think that would make things better. But I honestly think this is an uncomplicated problem and that this simple shell script is a fitting solution.

@silabs-hfegran, @MikeOpenHWGroup, this continues #1867

### CODE REVIEW

File: bin/clonetb
Comment (MikeOpenHWGroup) [inline bin/clonetb:73]: Two comments about this:
1. It would ease usability and maintenance if these were clearly visible as global vars at the top of the file.
2. Would it be possible to set things up so that if the user control these via shell env vars or command-line options?

File: bin/clonetb
Comment (MikeOpenHWGroup) [inline bin/clonetb:91]: Anytime you use `sed` or `awk` there should be some comments to explain extactly what is happening.  :stuck_out_tongue_closed_eyes:

**Review by MikeOpenHWGroup — CHANGES_REQUESTED** [review]

File: bin/clonetb
Comment (silabs-robin) [reply bin/clonetb:100]: Comments are now added, and each pipe is split to its own line.

File: bin/clonetb
Comment (silabs-robin) [reply bin/clonetb:79]: > It would ease usability and maintenance if these were clearly visible as global vars at the top of the file.

It would be the opposite of "progressive disclosure".
For normal usage there is no need to know, and for advanced usage it is trivial to figure it out.

Nevertheless, I've added a comment at the top of the file which explains the env vars.
It can be useful when modifying the script at a later time.

--------

> Would it be possible to set things up so that if the user control these via shell env vars or command-line options?

Having two ways of doing it?

I also played with that idea (and know a way to do it), but ultimately I think it adds unnecessary clutter.
Env vars like `CV_CORE` are already in use in core-v-verif.
It's nice and quick and clean and gets things done.

File: bin/clonetb
Comment (MikeOpenHWGroup) [reply bin/clonetb:100]: Good comments, thanks.

The resolution of a conversation is the responsibility of the individual who opened the conversation (in this case, me), or a Committer.

File: bin/clonetb
Comment (MikeOpenHWGroup) [reply bin/clonetb:79]: >  Having two ways of doing one thing?

We already have that.  The Makefiles are written such that you can either set vars on the command-line or as shell environment variables:
```
$ make test TEST=hello-world SIMULATOR=xrun
```
...is the same as...
```
$ export SIMULATOR=xrun
$ make test TEST=hello-world
```

Comment (silabs-robin) [reply bin/clonetb:79]: > We already have that. The Makefiles are written such that you can either set vars on the command-line or as shell environment variables
> [...]
> ...is the same as...
> [...]

Oh, I thought you were asking for command-line options, like `$ myprogram --myarg`.
Those two examples you gave are both using environment variables, not command-line options.
Our makefiles are not written to facilitate that, the syntax of make allows macro definitions to be given as arguments.
By default, posix (and gnu) shell allows you to pass environment variables by prepending them to a command, like `$ MYVAR=somevalue mycommand`, and the "Examples:" section of my comment header shows an example of this.

**Review by MikeOpenHWGroup — APPROVED** [review]

**Comment by MikeOpenHWGroup** [comment]
Approved.  I shall defer to @silabs-hfegran for final approval and merge.

**Review by silabs-hfegran — APPROVED** [review]

**Review by silabs-hfegran — APPROVED** [review]

**Pull request merged**
